specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,278 @@
+                            std::wstring message = L"Launching External Protocol request";
+                            if (initiating_uri.get() == L"")
+                            {
+                                message += L"from ";
+                                message += initiating_uri.get();
+                            }
+                            message += L" to ";
+                            message += uri.get();
+                            message += L"?\n\n";
+                            message += L"Do you want to grant permission?\n";
+                            int response = MessageBox(
+                                nullptr, message.c_str(), L"Launching External Protocol",
+                                MB_YESNOCANCEL | MB_ICONWARNING);
+                            if (response == IDYES)
+                            {
+                                args->put_Cancel(true);
+                                std::wstring protocol_url = L"calculator://";
+                                SHELLEXECUTEINFO info = {sizeof(info)};
+                                info.fMask = SEE_MASK_NOASYNC;
+                                info.lpVerb = L"open";
+                                info.lpFile = protocol_url.c_str();
+                                info.nShow = SW_SHOWNORMAL;
+                                ::ShellExecuteEx(&info);
+                            }
+                            else
+                            {
+                                args->put_Cancel(true);
+                            }
+                        }
+                        return S_OK;
+                    };
+                    wil::com_ptr<ICoreWebView2Deferral> deferral;
+                    CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                    m_appWindow->RunAsync(
+                        [deferral, showDialog]()
+                        {
+                            showDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_launchingExternalProtocolToken));
+    }
+}
+    
+```
+
+## .NET and WinRT
+
+```c#
+private WebView2 webView;
+void RegisterLaunchingExternalProtocolHandler() 
+{
+    webView.CoreWebView2.LaunchingExternalProtocol += (sender, args) {
+        {
+            CoreWebView2Deferral deferral = args.GetDeferral();
+            System.Threading.SynchronizationContext.Current.Post((_) =>
+            {
+                using (deferral)
+                {
+                    if (args.Uri == "calculator://")
+                    {
+                        // If this matches our desired protocol, then set the
+                        // event args to cancel the event and launch the
+                        // calculator app.
+                        Process.Start("calc");
```

Comment:
You should be launching args.Uri not 'calc' right?

Reply:
For Proccess.Start in .NET the protocol for calculator is only Calc and will fail to start the application if the args.URI is passed in. This is why I was considering keeping the handled property because for .NET the URI will not match the expected protocol needed.